resources/assets/js/features/trip/components/Details/TripRoutesPassengers.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { localize } from 'react-localize-redux';
+
+import TripRouteUser from './TripRouteUser';
+import { DriverIcon, SeatIcon } from 'app/components/Icons';
+
+
+class TripRoutesPassengers extends React.Component {
+
+    render() {
+        const { translate, maxSeats, routes, driver } = this.props;
+
+        return (
+            <section className="block-border p-3 mt-4">
+                <header className="trip-section-header">
+                    <h3 className="h5">
+                        { translate('trip_details.routes_passengers.header') }
+                    </h3>
+                </header>
+
+                <div className="table-responsive">
+                    <table className="table trip-routes">
+                        <thead>
+                          <tr>
+                            <th />
+                            <th>
+                                <DriverIcon className="trip-routes__head-icon" />
+                            </th>
+                        {
+                          [...Array(maxSeats)].map((n, i) =>
+                            <th key={i}>
+                                <SeatIcon className="trip-routes__head-icon" />
+                            </th>
+                          )
+                        }
+                          </tr>
+                        </thead>
+                        <tbody>
+                        {
+                          routes.map((route, i) =>
+                            <tr key={i}>
```

Comment:
render is too big, split this on components and methods that will return results, avoid using i === 0 in template, you can pass this as param like showSomething: true/false in new component from this one based on index